.github/workflows/nim-ci.yml

```diff
@@ -0,0 +1,38 @@
+name: Nim Compilability CI
+
+on:
+  push:
+    paths:
+      - "**.nim"
+  pull_request:
+    paths:
+      - "**.nim"
+
+jobs:
+  check-compilation:
+    runs-on: ubuntu-latest
+    container: nimlang/nim:1.2.6
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
+
+      - name: Retrieve changed files
+        id: getfiles
+        run: |
+          echo "::set-output name=files::$(git diff-tree --no-commit-id --diff-filter=d --name-only -r ${{ github.sha }} | xargs)"
+        shell: bash
```

Comment:
I'm not sure this works in all cases. What if you have a PR with 3 commits? Then you would not find out about any of the changes in the first commit. We've struggled with this too for our other workflows. For pull requests, you can do [this](https://github.com/exercism/v3/blob/master/.github/workflows/verify-code-formatting.yml#L21). 

This would of course not work for the master push.

I'm wondering if maybe you should just check _all_ the Nim files for the `master` push. I think the check command is not that performance intensive, is it?

Reply:
So given the context dump that is shown in the GHA runs of the past couple commits, `github.event.pull_request.base.sha` & `github.event.before` should be useful over here in getting the changed files since the last push in the context of a commit and getting the files since before the PR in the context of a PR. This should solve the issue in getting the appropriate files

Reply:
Yep, this looks like it should work.